src/core/dataset.ts

```diff
@@ -90,5 +104,23 @@ export class Dataset {
       return this;
     }
   }
+
+  public keyFunction(): (datum: any, index: number) => any;
+  /**
+   * Sets the key.
```

Comment:
"Sets the keyFunction."

Also can you explain what a keyFunction is here in a very brief summary (1-3 lines)